lib/Runtime/Types/DictionaryTypeHandler.cpp

```diff
@@ -924,11 +928,20 @@ namespace Js
             "Numeric property names should have been converted to uint or PropertyRecord* ");
 
         ScriptContext* scriptContext = instance->GetScriptContext();
+        JavascriptLibrary* library = scriptContext->GetLibrary();
         DictionaryPropertyDescriptor<T>* descriptor;
         JsUtil::CharacterBuffer<WCHAR> propertyName(propertyNameString->GetString(), propertyNameString->GetLength());
 
         if (propertyMap->TryGetReference(propertyName, &descriptor))
         {
+            if (!this->GetHasSpecialProperties() && NoSpecialPropertyCache::IsDefaultHandledSpecialProperty(propertyNameString))
+            {
+                this->SetHasSpecialProperties();
```

Comment:
I don't really understand what we're doing here. Why would deleting a property lead to setting the special-properties flag?

Reply:
If you are deleting a valueOf/toString and the flag wasn’t set, it means you are deleting it off of Object.prototype, and so you will have non-standard behavior if they are accessed. I’ll add a comment.